crates/core/tedge_mapper/src/collectd/mapper.rs

```diff
@@ -18,6 +18,16 @@ impl CollectdMapper {
 
 #[async_trait]
 impl TEdgeComponent for CollectdMapper {
+    fn session_name(&self) -> &str {
+        COLLECTD_MAPPER_NAME
+    }
+
+    async fn init(&self) -> Result<(), anyhow::Error> {
+        info!("Initialize tedge mapper collectd");
+        mqtt_channel::init_session(&self.get_mqtt_config()?).await?;
+        Ok(())
+    }
```

Comment:
This is buggy. Since the configuration returned by `self.get_mqtt_config()` doesn't set the list of subscriptions for collectd.

It would be better to introduce a  `TEdgeComponent::init_session` reference implementation and to use it here.

```suggestion
    async fn init(&self) -> Result<(), anyhow::Error> {
        self.init_session(collectd_topics).await?
    }
```